lnwallet/rpcwallet/remote_signer_client.go

```diff
@@ -0,0 +1,959 @@
+//
+// NOTE: This is part of the SignCoordinatorStreamFeeder interface.
+func (s *StreamFeeder) Stop() {
+	close(s.quit)
+
+	s.wg.Wait()
+}
+
+// GetStream returns a new stream to the watch-only node, by making an
+// outbound gRPC connection to the watch-only node. The function also returns a
+// cleanup function that closes the connection, which should be called when the
+// stream is no longer needed.
+//
+// NOTE: This is part of the SignCoordinatorStreamFeeder interface.
+func (s *StreamFeeder) GetStream(streamCtx context.Context) (
+	StreamClient, func(), error) {
+
+	select {
+	// Don't run if the StreamFeeder has already been shutdown.
+	case <-s.quit:
+		return nil, nil, ErrShuttingDown
+	default:
+	}
+
+	// Create a new outbound gRPC connection to the watch-only node.
+	conn, err := s.getClientConn()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	cleanUp := func() {
+		conn.Close()
+	}
+
+	// Wrap the connection in a WalletKitClient.
+	walletKitClient := walletrpc.NewWalletKitClient(conn)
+
+	// Create a new stream to the watch-only node.
+	stream, err := walletKitClient.SignCoordinatorStreams(streamCtx)
+	if err != nil {
+		cleanUp()
+
+		return nil, nil, err
+	}
+
+	return stream, cleanUp, nil
+}
+
+// getClientConn creates a new outbound gRPC connection to the watch-only node.
+func (s *StreamFeeder) getClientConn() (*grpc.ClientConn, error) {
+	// If we fail to connect to the watch-only node within the
+	// configured timeout we should return an error.
+	ctxt, cancel := context.WithTimeout(
+		context.Background(), s.timeout,
+	)
+	defer cancel()
+
+	// Load the specified macaroon file for the watch-only node.
+	macBytes, err := os.ReadFile(s.macaroonPath)
+	if err != nil {
+		return nil, fmt.Errorf("could not read macaroon file: %w", err)
+	}
+
+	mac := &macaroon.Macaroon{}
+
+	err = mac.UnmarshalBinary(macBytes)
+	if err != nil {
+		return nil, fmt.Errorf("could not unmarshal macaroon: %w", err)
+	}
+
+	macCred, err := macaroons.NewMacaroonCredential(mac)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"could not create macaroon credential: %w", err)
+	}
+
+	// Load the specified TLS cert for the watch-only node.
+	tlsCreds, err := credentials.NewClientTLSFromFile(s.tlsCertPath, "")
+	if err != nil {
+		return nil, fmt.Errorf("could not load TLS cert: %w", err)
+	}
+
+	opts := []grpc.DialOption{
+		grpc.WithBlock(),
+		grpc.WithTransportCredentials(tlsCreds),
+		grpc.WithPerRPCCredentials(macCred),
+	}
+
+	var (
+		// A channel to signal when has successfully been created.
+		connDoneChan = make(chan *grpc.ClientConn)
+		errChan      = make(chan error)
+	)
+
+	// Now let's try to connect to the watch-only node. We'll do this in a
+	// goroutine to ensure we can exit if the quit channel is closed. If the
+	// quit channel is closed, the context will also be canceled, hence
+	// stopping the goroutine.
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+
+		conn, err := grpc.DialContext(ctxt, s.rpcHost, opts...)
```

Comment:
I think we should also add an info level log here that we're attempting a connection (and what host:port we're connecting to).

Reply:
Done :)